parquet/src/test/java/org/apache/iceberg/parquet/TestParquetEncryption.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.parquet;
+
+import static org.apache.iceberg.Files.localInput;
+import static org.apache.iceberg.Files.localOutput;
+import static org.apache.iceberg.parquet.ParquetWritingTestUtils.createTempFile;
+import static org.apache.iceberg.types.Types.NestedField.optional;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Random;
+import org.apache.avro.generic.GenericData;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.TestHelpers;
+import org.apache.iceberg.avro.AvroSchemaUtil;
+import org.apache.iceberg.io.CloseableIterator;
+import org.apache.iceberg.io.FileAppender;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.types.Types.IntegerType;
+import org.apache.parquet.crypto.ParquetCryptoRuntimeException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class TestParquetEncryption {
```

Comment:
This is testing the old read/write path, not the new one. The old one uses Parquet's `ReadSupport`, which is no longer used in the engine integrations.

This is okay to have so we know that the Parquet stuff works, but it doesn't test what we actually use. I think we need a copy of that calls `createReaderFunc` and `createBatchedReaderFunc`, as well as the corresponding writer option (you only need to write once, there is no batched write path).

For a `createReaderFunc` example, see `TestGenericData` in the `parquet` module. Then you can add tests that call `createBatchedReaderFunc` based on `TestParquetVectorizedReads`.

Reply:
I've created a unitest for encryption with `createReaderFunc` , based on the `TestGenericData`. For the `createBatchedReaderFunc`, I've added a small-footprint code to the `TestParquetVectorizedReads`, which adds encryption coverage to all usecases tested in that class.

Reply:
Btw, in a later PR, we have encryption unitests with Spark SQL (that run with the `BaseBatchReader` in their stack), so they also test the `createBatchedReaderFunc` with encryption.

Reply:
The new generic test looks good to me.

I think this should be `TestParquetEncryptionWithWriteSupport`